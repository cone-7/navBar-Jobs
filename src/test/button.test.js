import React from 'react';
import ButtonComponent from '../components/button.js';
import renderer from 'react-test-renderer';

test('render button', () => {
  const component = renderer.create(
    <ButtonComponent >Facebook</ButtonComponent>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

test('clicked button', () => {
  const component = renderer.create(
  <ButtonComponent onClick={() => console.log('button')}></ButtonComponent>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
