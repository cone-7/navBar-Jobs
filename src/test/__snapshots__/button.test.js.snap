// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicked button 1`] = `
<button
  onClick={[Function]}
/>
`;

exports[`clicked button 2`] = `
<button
  onClick={[Function]}
/>
`;

exports[`render button 1`] = `
<button
  onClick={[Function]}
>
  Facebook
</button>
`;
